src/config.hpp

```diff
@@ -308,7 +308,7 @@ class config
 		bool operator==(const const_attribute_iterator &i) const { return i_ == i.i_; }
 		bool operator!=(const const_attribute_iterator &i) const { return i_ != i.i_; }
 		bool operator==(const attribute_iterator &i) const { return i_ == i.i_; }
-		bool operator!=(const attribute_iterator &i) const { return i_ != i.i_; }
+		bool operator!=(const attribute_iterator &i) const { return !operator==(i); }
```

Comment:
While all this is probably OK, I consider it a code-smell they the entire struct encapsulates an iterator, instead of inherits, and, thus, must do all this boilerplate. 

I'm not interested in digging into the class/struct .. just saying it may be a lot of effort for no real effect and a typedef might do the same thing.